imap_processing/codice/codice_l1a.py

```diff
@@ -719,6 +789,75 @@ def get_params(dataset: xr.Dataset) -> tuple[int, int, int, int]:
     return table_id, plan_id, plan_step, view_id
 
 
+def group_data(packets: xr.Dataset) -> list[bytes]:
+    """
+    Organize continuation packets into appropriate groups.
+
+    Some packets are continuation packets, as in, they are packets that are
+    part of a group of packets. These packets are marked by the `seq_flgs` field
+    in the CCSDS header of the packet. For CoDICE, the values are defined as
+    follows:
+
+    3 = Packet is not part of a group
+    1 = Packet is the first packet of the group
+    0 = Packet is in the middle of the group
+    2 = Packet is the last packet of the group
+
+    For packets that are part of a group, the byte count associated with the
+    first packet of the group signifies the byte count for the entire group.
+
+    Parameters
+    ----------
+    packets : xarray.Dataset
+        Dataset containing the packets to group.
+
+    Returns
+    -------
+    grouped_data : list[bytes]
+        The packet data, converted to bytes and grouped appropriately.
+    """
+    grouped_data = []  # Holds the properly grouped data to be decompressed
+    current_group = bytearray()  # Temporary storage for current group
+    group_byte_count = None  # Temporary storage for current group byte count
+
+    for packet_data, group_code, byte_count in zip(
+        packets.event_data.data, packets.seq_flgs.data, packets.byte_count.data
+    ):
+        # Convert from numpy array to byte object
+        compressed_packet_values = ast.literal_eval(str(packet_data))
```

Comment:
Why do you need to do this step with literal_eval? Are you trying to get to integers, do you need to call `.to_bytes()` on the result `int(packet_data, 2).to_bytes()?
If you have a numpy array you should be able to call `tobytes()` https://numpy.org/doc/2.2/reference/generated/numpy.ndarray.tobytes.html But, if I remember right, I think you have a string dtype with 1s and 0s representing the binary? So can you append all those 1s and 0s together as a string and do this in one final step?

Reply:
Thanks for catching this! I think this is some sort of leftover code from an earlier implementation, because it turns out this step is not needed at all. `packet_data` is already a `bytes` type, so I can just directly append other grouped data and/or cut out the padding on that variable.